### PR TITLE
Temp

### PR DESCRIPTION
**Pull request by Zonespace27**

### CODE REVIEW

File: client/client.py [hunk 1]
```diff
@@ -0,0 +1,272 @@
+import socket
```
Comment (g15h) [inline client/client.py:1]: Alphabetize

File: client/client.py [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+import pickle
+import string
+import curses
+import constants as const
+import time
+from math import trunc
+
+class Client():
+
+    def recieve_messages(self, socket_class: socket.socket):
```
Comment (g15h) [inline client/client.py:16]: Function comment

File: client/client.py [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+import sys
+import re
+import pickle
+import string
+import curses
+import constants as const
+import time
+from math import trunc
+
+class Client():
```
Comment (g15h) [inline client/client.py:14]: Class comment

File: client/client.py [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+
+
+    def line_text_multiline_check(self, text_to_check: str) -> list:
+        """
+        Slightly misleading name. Adds calls add_line_text after checking
+        if the text wouldn't fit in one line, and splits it out into multiple messages if so.
+        """
+        list_of_lines = []
+        if len(text_to_check) > self.x_size:
+            amount_of_lines = trunc(len(text_to_check) / self.x_size)
```
Comment (g15h) [inline client/client.py:40]: line_count would be more clear

File: client/client.py [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+
+            for i in range(amount_of_lines):
+                sliced_text = text_to_check[(self.x_size * i):(self.x_size * min(i + 1, amount_of_lines))] # Slices the string within certain bounds determined by line count
+                list_of_lines.append(sliced_text)
+                self.add_line_text(sliced_text)
+            
+            # For anything that was rounded off
+            sliced_text = text_to_check[(amount_of_lines * self.x_size):]
+            list_of_lines += sliced_text
```
Comment (g15h) [inline client/client.py:49]: use .append() instead of +=

File: client/client.py [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+            self.add_line_text(sliced_text)
+        
+        else:
+            list_of_lines += text_to_check
```
Comment (g15h) [inline client/client.py:53]: .append()

File: client/client.py [hunk 7]
```diff
@@ -0,0 +1,272 @@
...
+            self.add_line_text(text_to_check)
+        
+        return list_of_lines
+
+
+    def add_line_text(self, text_to_add: str):
+        """
+        Adds text to the screen, in addition to adding it to the `line_text` list with a timestamp. \n
```
Comment (g15h) [inline client/client.py:61]: Is the \n necessary?

File: client/client.py [hunk 8]
```diff
@@ -0,0 +1,272 @@
...
+            
+        self.main_screen.refresh()
+    
+
+    def move_up_lines(self, move_amount: int):
+        """
+        Move up the lines on the screen by `move_amount` lines.
+        Moved messages are currently entirely deleted.
+        """
+        line_text_keys = list(self.line_text.keys())
```
Comment (g15h) [inline client/client.py:80]: don't store sequential data in a dictionary, use a list

File: client/client.py [hunk 9]
```diff
@@ -0,0 +1,272 @@
...
+        line_text_values = list(self.line_text.values())
+        line_text_keys = line_text_keys[move_amount:] # Might add something for seeing history later, currently just shadow realms old messages
+        line_text_values = line_text_values[move_amount:]
+        self.line_text = {}
```
Comment (g15h) [inline client/client.py:84]: All member variables should be declared in _ _ init _ _(self)

File: client/client.py [hunk 10]
```diff
@@ -0,0 +1,272 @@
...
+        for key in line_text_keys:
```
Comment (g15h) [inline client/client.py:85]: Again, use a list instead of a dictionary

File: client/client.py [hunk 11]
```diff
@@ -0,0 +1,272 @@
...
+        As such, we need to do per-character handling, which this function does.
+        """
+        # If it's text, handle it early
+        ascii_char = chr(character)
+        try:
+            if self.valid_characters.index(ascii_char) or ascii_char == "0":
+                if len(self.inputted_characters) >= (self.x_size * const.INPUT_LINE_AMT):
+                    return False
+
+                self.inputted_characters += ascii_char
```
Comment (g15h) [inline client/client.py:131]: declare in init()

File: client/constants.py [hunk 12]
```diff
@@ -0,0 +1,12 @@
+# constants operate from the bottom-right, to the top-left.
```
Comment (g15h) [inline client/constants.py:1]: If this is only referenced in one file, you could consolidate it

File: server/admin_commands.py [hunk 13]
```diff
@@ -0,0 +1,60 @@
+from json import load
+import file_locations_server as file_locations
+import client_commands as cli
```
Comment (g15h) [inline server/admin_commands.py:3]: alphabetize

File: server/client_commands.py [hunk 14]
```diff
@@ -0,0 +1,76 @@
...
+#### Below are the equivalent of commands in function form for readability
+
+# Displays commands
+def command_help(executor, server_class, argument):
+    compiled_string = ''
+    for command_name in server_class.name_to_command:
+        command_class = server_class.name_to_command.get(command_name)
+        if command_class.admin_only:
+            continue
+        compiled_string += command_class.name + ': ' + command_class.desc + (', ' if command_name != list(server_class.name_to_command)[len(server_class.name_to_command) - 1] else '')
```
Comment (g15h) [inline server/client_commands.py:19]: try to reformat this so it's not so long

File: server/exec.py [hunk 15]
```diff
@@ -0,0 +1,10 @@
+# Allows early returning of `exec()`, use `Exec` over `exec` function
```
Comment (g15h) [inline server/exec.py:1]: You can put module comments in """ """ at top of file

File: server/file_locations_server.py [hunk 16]
```diff
@@ -0,0 +1,6 @@
+CLIENT_COMMANDS_FILE = 'client_command.json'
```
Comment (g15h) [inline server/file_locations_server.py:1]: try to consolidate if only used in one file

File: server/message_handler.py [hunk 17]
```diff
@@ -0,0 +1,108 @@
+import re
+from exec import Exec
+import time
+import threading
+import time
```
Comment (g15h) [inline server/message_handler.py:5]: duplicate

File: server/server.py [hunk 18]
```diff
@@ -0,0 +1,350 @@
+import socket
```
Comment (g15h) [inline server/server.py:1]: alphabetize

File: server/server.py [hunk 19]
```diff
@@ -0,0 +1,350 @@
...
+import sys
+import random
+import string
+import message_handler as messageHandler
+
+class Server():
+    HOST = "127.0.0.1" #"127.0.0.1"
+    PORT = 65432 #65432
+
+    all_connections = []
```
Comment (g15h) [inline server/server.py:18]: you can add type hints for these

File: client/client.py (hunk 7 above)
Comment (Zonespace27) [reply client/client.py:61]: image
I prefer it looking like this instead of being all in one line

File: client/client.py (hunk 9 above)
Comment (Zonespace27) [reply client/client.py:84]: It is, just init's at the bottom (moved now)

File: client/client.py (hunk 11 above)
Comment (Zonespace27) [reply client/client.py:131]: same story

File: client/constants.py (hunk 12 above)
Comment (Zonespace27) [reply client/constants.py:1]: I do want this separate for readability purposes

File: server/client_commands.py (hunk 14 above)
Comment (Zonespace27) [reply server/client_commands.py:19]: Can't, ultimately; as the comment says, these are just for a dev to understand how the commands work, none of these functions are ever executed. It's a requirement that you should just be able to copy+paste this, add the newlines and indents, and it'll work in exec()

File: server/file_locations_server.py (hunk 16 above)
Comment (Zonespace27) [reply server/file_locations_server.py:1]: same story as others

File: server/server.py (hunk 19 above)
Comment (Zonespace27) [reply server/server.py:18]: Python actually silently assigns type hints if you declare it to be a primitive type on creation

File: client/client.py (hunk 8 above)
Comment (Zonespace27) [reply client/client.py:80]: cleaned up

**Pull request merged**